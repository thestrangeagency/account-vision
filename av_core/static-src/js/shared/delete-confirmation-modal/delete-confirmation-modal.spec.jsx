import React from 'react';
import { shallow } from 'enzyme';

import DeleteConfirmationModal from './delete-confirmation-modal';

test('it renders', () => {
  const wrapper = shallow(<DeleteConfirmationModal onClickConfirm={() => {}} />);
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ isActionDisabled: true });
  expect(wrapper).toMatchSnapshot();
});

test('it calls the confirm function prop', () => {
  const onClickConfirm = jest.fn();
  const wrapper = shallow(<DeleteConfirmationModal onClickConfirm={onClickConfirm} />);
  wrapper.find('.btn-danger').simulate('click');
  expect(onClickConfirm).toHaveBeenCalled();
});
