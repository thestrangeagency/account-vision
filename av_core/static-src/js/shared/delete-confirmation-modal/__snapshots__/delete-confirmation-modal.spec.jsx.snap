// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  aria-hidden="true"
  className="modal fade"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-body"
      >
        <h5>
          Are you sure?
        </h5>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders 2`] = `
<div
  aria-hidden="true"
  className="modal fade"
  id="confirmDeleteModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-body"
      >
        <h5>
          Are you sure?
        </h5>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          disabled={true}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
